syntax = "proto3";

package authentication;

service Auth {
    rpc VerifyToken (VerifyTokenRequest) returns (VerifyTokenResponse);
}

message VerifyTokenRequest {
    string accessToken = 1;
}

message VerifyTokenResponse {
    bool isValid = 1;
    User user = 2;
}

message User {
    string userId = 1;
    string username = 2;
    string role = 3;
}